### ISSUE TITLE
Issue with local diff on NFS mount with symlinks

### ISSUE THREAD

**Issue by jimbair** [juhp/pkgtreediff#2]
While diffing two composes that are stored on NFS (with symlinks back to the packages to avoid duplicate rpms across composes), it appears to be skipping over the diff entirely. I assume either the target folders being on NFS, or the packages in the folders being symlinks, is the culprit.

* When run locally on the server, it cleanly exits (0) in 0.003s with no diff results
* When run over http from my laptop aimed at the same server, the diff takes 1m40s with 44 updates showing

I'll send an email as well with some more details.

**Comment by jimbair** [comment]
Update: I fetched two full composes from the same branch, and local diffs seem to not run at all? These same two folders work great over HTTP:


```

[jim@lenovo test]$ ls
17  3
[jim@lenovo test]$ du -hscL *
36G	17
36G	3
72G	total
[jim@lenovo test]$ time pkgtreediff ./3 ./17 -r -A

real	0m0.006s
user	0m0.004s
sys	0m0.002s
[jim@lenovo test]$ echo $?
0
[jim@lenovo test]$ 

```

This is running on Fedora 30 (laptop) and centos 7 (server) using copr repos as well with latest versions in copr.

**Comment by juhp** [comment]
I suspect it may be ignoring symlinks, but that can surely be addressed.

**Comment by jimbair** [comment]
Yep, that was my assumption as well! I appreciate you looking into it.

**Comment by juhp** [comment]
I can't really see anything yet in my code that could be affecting this.
Could you give some example of the symlinks you have?

Maybe I need to pass you a test prog or find a machine with nfs.
